src/api/spec/models/workflow/step/set_flags_step_spec.rb

```diff
@@ -0,0 +1,233 @@
+                  repository: 'openSUSE_Tumbleweed',
+                  architecture: 'x86_64'
+                },
+                {
+                  type: 'publish',
+                  status: 'enable',
+                  project: 'home:Iggy'
+                }
+              ]
+          }
+        end
+
+        it 'add flags to the project' do
+          expect { subject.call }.to change(Flag, :count).by(2)
+          expect(Flag.all).to match_array([
+                                            have_attributes(status: 'enable', repo: 'openSUSE_Tumbleweed', project_id: target_project.id, package_id: nil, flag: 'build'),
+                                            have_attributes(status: 'enable', project_id: target_project.id, package_id: nil, flag: 'publish')
+                                          ])
+        end
+      end
+    end
+
+    context 'when user have the permission and the package is valid' do
+      let!(:target_package) { create(:package, project: user.home_project, name: 'ctris-0087aa5c0549a6cc0b4c1bb324d2fa8dc665e063') }
+      let(:payload) { { scm: 'gitlab', event: 'Push Hook', commit_sha: '0087aa5c0549a6cc0b4c1bb324d2fa8dc665e063' } }
+      let(:scm_webhook) { ScmWebhook.new(payload: payload) }
+      let(:step_instructions) do
+        {
+          flags:
+            [
+              {
+                type: 'lock',
+                status: 'disable',
+                project: 'home:Iggy',
+                package: 'ctris'
+              }
+            ]
+        }
+      end
+
+      before do
+        # rubocop:disable RSpec/AnyInstance
+        allow_any_instance_of(Package).to receive(:commit_user).and_return(user)
+        # rubocop:enable RSpec/AnyInstance
+      end
+
+      it 'add flags to the package' do
+        expect { subject.call }.to change(Flag, :count).by(1)
+        expect(Flag.all).to match_array([
+                                          have_attributes(status: 'disable', project_id: nil, package_id: target_package.id, flag: 'lock')
+                                        ])
+      end
+    end
+
+    context 'when there is a duplicate flag' do
+      let(:scm_webhook) do
+        ScmWebhook.new(payload: {
+                         scm: 'github',
+                         event: 'pull_request',
+                         action: 'opened',
+                         pr_number: 1,
+                         target_repository_full_name: 'openSUSE/repo123',
+                         commit_sha: '123'
+                       })
+      end
+      let(:step_instructions) do
+        {
+          flags:
+            [
+              {
+                type: 'build',
+                status: 'enable',
+                project: 'home:Iggy',
+                repository: 'openSUSE_Tumbleweed',
+                architecture: 'x86_64'
+              }
+            ]
+        }
+      end
+
+      before do
+        target_project.add_flag('build', 'enable', 'openSUSE_Tumbleweed', 'x86_64')
```

Comment:
It could become a factory. If you look on `spec/factories/project.rb` we have factories for `forbidden_project` and `locked_project`.. we could for sure have one `build_enabled_project`

Reply:
That's right we can have a new factory but we don't need one in this case because I have a single instance of `target_project`. I need the flag in only one example.